vars/pullGitRepositoryToSapCloudPlatformAbapEnvironment.groovy

```diff
@@ -0,0 +1,213 @@
+     */
+    'host',
+    /**
+     * Specifies the name of the Repository (Software Component) on the SAP Cloud Platform ABAP Environment system
+     */
+    'repositoryName'
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    /**
+     * Specifies the communication user of the communication scenario SAP_COM_0510
+     */
+    'username',
+    /**
+     * Specifies the password of the communication user
+     */
+    'password'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+/**
+ * Pulls a Repository (Software Component) to a SAP Cloud Platform ABAP Environment system.
+ *
+ * !!! note "Git Repository and Software Component"
+ *       In SAP Cloud Platform ABAP Environment Git repositories are wrapped in Software Components (which are managed in the App "Manage Software Components")
+ *       Currently, those two names are used synonymous.
+ * !!! note "User and Password"
+ *        In the future, we want to support the user / password creation via the create-service-key funcion of cloud foundry.
+ *        For this case, it is not possible to use the usual pattern with Jenkins Credentials.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters, failOnError: true) {
+
+        def script = checkScript(this, parameters) ?: this
+
+        // In the future, we want to support the user / password creation via the create-service-key funcion of cloud foundry.
+        // For this case, it is not possible to use the usual pattern with Jenkins Credentials.
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .withMandatoryProperty('host', 'Host not provided')
+            .withMandatoryProperty('repositoryName', 'Repository / Software Component not provided')
+            .withMandatoryProperty('username')
+            .withMandatoryProperty('password')
+            .collectValidationFailures()
+            .use()
+
+        if (!(configuration.host =~ /^(https:\/\/)(.*)/).find()) {
+            error "[${STEP_NAME}] URL Validation Failed: HTTPS must be used"
+        }
+
+        String usernameColonPassword = configuration.username + ":" + configuration.password
+        String authToken = usernameColonPassword.bytes.encodeBase64().toString()
+        String urlString = configuration.host + '/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull'
+        echo "[${STEP_NAME}] General Parameters: URL = \"${urlString}\", repositoryName = \"${configuration.repositoryName}\""
+        HeaderFiles headerFiles = new HeaderFiles()
+
+        try {
+            String urlPullEntity = triggerPull(configuration, urlString, authToken, headerFiles)
```

Comment:
[just curiosity] what does it mean if `urlPullEntity` is null? Is it OK to silently finish the step?

Reply:
should not happen - but I'll add some error handling